File: java/src/sdk/dataobjects/paypal/payflow/ACHTender.java
package paypal.payflow;

/**
 * This class is used to create and use an ACH( Automatic Clearing House ) Tender type.
 * BankAcct is the Payment device associated with this tender type. {@link BankAcct}
 */

public final class ACHTender extends BaseTender {

    /**
     * ACH Auth Type
     */
    private String authType;
    /**
     * ACH Prenote (Y/N)
     */
    private String preNote;
    /**
     * ACH Term City
     */
    private String termCity;
    /**
     * ACH Term State
     */
    private String termState;

    /**
     * ACHTender should be used to perform the transactions
     * in which the user provides his bank account details for
     * the online payment processing.
     *
     * @param bankAccount String
     *  Maps to Payflow Parameter: TENDER
     *  * <p>
     * .............
     * //bnkAcct is the populated BankAcct object.
     * .............
     *  * <p>
     * //Create the Tender object
     * ACHTender tender = new ACHTender(bnkAcct);
     */
    public ACHTender(BankAcct bankAccount) {
        super(PayflowConstants.TENDERTYPE_ACH, bankAccount);
    }

    /**
     * Gets the Authtype.
     *  * <p>
     * The type of authorization received from the payer.
     * Allowed Auth Types are:
     *
     * AUTHTYPE - Description
     *
     * CCD - Default for B2B format accounts
     * PPD - Standard customer authorization method) for B2C format accounts.
     * ARC - Accounts Receivables check entry for a single entry debit.
     * RCK - Re-presented check entry for a single entry debit.
     * WEB - The customer authorized the payment over the Internet.
     * TEL - Debit authorization obtained by telephone.
     * POP - Point of Purchase check entry for a single entry debit.
     *
     * @return authType
     *  <p> Maps to Payflow Parameter: AUTHTYPE </p>
     */

    public String getAuthType() {
        return authType;
    }

    /**
     * Sets the Authtype.
     *  * <p>
     * The type of authorization received from the payer.
     * Allowed Auth Types are:
     *
     * AUTHTYPE - Description
     *
     * CCD - Default for B2B format accounts
     * PPD - Standard customer authorization method) for B2C format accounts.
     * ARC - Accounts Receivables check entry for a single entry debit.
     * RCK - Re-presented check entry for a single entry debit.
     * WEB - The customer authorized the payment over the Internet.
     * TEL - Debit authorization obtained by telephone.
     * POP - Point of Purchase check entry for a single entry debit.
     *
     * @param authType String
     *  <p> Maps to Payflow Parameter: AUTHTYPE </p>
     */
    public void setAuthType(String authType) {
        this.authType = authType;
    }

    /**
     * Gets the PreNote.
     *  * <p>
     * Prenote indicates a prenotification payment with no amount.
     * Used to verify bank account validity. Receiving banks are not required
     * to respond to prenotification payments.
     * Allowed prenote values are:
     *
     * PRENOTE  - Description
     *
     * N Default. AMT needs to be passed.
     * Y Default. AMT does not need to be passed.
     *
     * @return preNote
     *  <p> Maps to Payflow Parameter: PRENOTE </p>
     */
    public String getPreNote() {
        return preNote;
    }

    /**
     * Sets the PreNote.
     *  * <p>
     * Prenote indicates a prenotification payment with no amount.
     * Used to verify bank account validity. Receiving banks are not required
     * to respond to prenotification payments.
     * Allowed prenote values are:
     *
     * PRENOTE  - Description
     *
     * N Default. AMT needs to be passed.
     * Y Default. AMT does not need to be passed.
     *
     * @param preNote String
     *  <p> Maps to Payflow Parameter: PRENOTE </p>
     */
    public void setPreNote(String preNote) {
        this.preNote = preNote;
    }

    /**
     * Gets the Term City.
     * <p>City where the merchant's terminal is located.
     * Used only for POP.</p>
     *
     * @return termCity
     *  <p> Maps to Payflow Parameter: TERMCITY </p>
     */
    public String getTermCity() {
        return termCity;
    }

    /**
     * Gets the Term City.
     * <p>City where the merchant's terminal is located.
     * Used only for POP.</p>
     *
     * @param termCity String
     *  <p> Maps to Payflow Parameter: TERMCITY </p>
     */
    public void setTermCity(String termCity) {
        this.termCity = termCity;
    }

    /**
     * Gets the TermState.
     * <p>State where the merchant's terminal is located.
     * Used only for POP.</p>
     *
     * @return termState
     *  <p> Maps to Payflow Parameter: TERMSTATE </p>
     */
    public String getTermState() {
        return termState;
    }

    /**
     * Gets the TermState.
     * <p> State where the merchant's terminal is located.
     * Used only for POP.</p>
     *
     * @param termState String
     *  <p> Maps to Payflow Parameter: TERMSTATE </p>
     */
    public void setTermState(String termState) {
        this.termState = termState;
    }

    protected void generateRequest() {
        super.generateRequest();
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_AUTHTYPE, authType));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_PRENOTE, preNote));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_TERMCITY, termCity));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_TERMSTATE, termState));
    }


}

File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/ACHTender.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// This class is used to create and use an ACH
	/// ( Automatic Clearing House ) Tender type.
	/// </summary>
	/// <remarks>BankAcct is the Payment device associated with this
	/// tender type.</remarks>
	/// <seealso cref="BankAcct"/>
	public sealed class ACHTender : BaseTender
	{
        #region "Member Variables"

        /// <summary>
        /// ACH Auth Type
        /// </summary>
        private String mAuthType;

        /// <summary>
        /// ACH Prenote (Y/N)
        /// </summary>
        private String mPreNote;

		/// <summary>
		/// ACH Term City
		/// </summary>
		private String mTermCity;

		/// <summary>
		/// ACH Term State
		/// </summary>
		private String mTermState;

		#endregion 

		#region "Constructors"

		/// <summary>
		/// Constructor
		/// </summary>
		/// <param name="BnkAcct">Bank Account object.</param>
		/// <remarks>
		/// ACHTender should be used to perform the transactions 
		/// in which the user provides his bank account details for
		/// the online payment processing.
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>TENDER</code>
		/// </remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		///		.............
		///		//BnkAcct is the populated BankAcct object.
		///		.............
		///		
		///		//Create the Tender object
		///		ACHTender Tender = new ACHTender(BnkAcct);
		///		
		///		.............
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///		.............
		///		//BnkAcct is the populated BankAcct object.
		///		.............
		///		
		///		'Create the Tender object
		///		Dim Tender As ACHTender = new ACHTender(BnkAcct)
		///		
		///		.............
		/// </code>
		/// <seealso cref="BankAcct"/>
		/// </example>
		public ACHTender(BankAcct BnkAcct) : base(PayflowConstants.TENDERTYPE_ACH, BnkAcct)
		{
		}

		#endregion

		#region "Properties"

		/// <summary>
		/// Gets, Sets the Prenote.
		/// </summary>
		/// <remarks>
		/// <para>Prenote indicates a prenotification payment with no amount.
		/// Used to verify bank account validity. Receiving banks are not required 
		/// to respond to prenotification payments.</para>
		/// <para>Allowed Prenote values are:</para>
		/// <list type="table">
		/// <listheader>
		/// <term>PRENOTE</term>
		/// <description>Description</description>
		/// </listheader>
		/// <item>
		/// <term>N</term>
		/// <description>Default. AMT needs to be passed.</description>
		/// </item>
		/// <item>
		/// <term>Y</term>
		/// <description>Default. AMT does not need to be passed.</description>
		/// </item>
		/// </list>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>PRENOTE</code>
		/// </remarks>
		public String PreNote
		{
			get { return mPreNote; }

			set { mPreNote = value; }
		}

		/// <summary>
		/// Gets, Sets the Termcity.
		/// </summary>
		/// <remarks>
		/// <para>City where the merchant's terminal is located. 
		/// Used only for POP.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>TERMCITY</code>
		/// </remarks>
		public String TermCity
		{
			get { return mTermCity; }

			set { mTermCity = value; }
		}

		/// <summary>
		/// Gets, Sets the Termstate.
		/// </summary>
		/// <remarks>
		/// <para>State where the merchant's terminal is located. 
		/// Used only for POP.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>TERMSTATE</code>
		/// </remarks>
		public String TermState
		{
			get { return mTermState; }

			set { mTermState = value; }
		}

		
		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			try
			{
				base.GenerateRequest();
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_AUTHTYPE, mAuthType));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_PRENOTE, mPreNote));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_TERMCITY, mTermCity));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_TERMSTATE, mTermState));
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				DataObjectException DEx = new DataObjectException(Ex);
				throw DEx;
			}
		}

		#endregion


	}
}